cardano-sync/src/Cardano/Sync/LedgerState.hs

```diff
@@ -222,6 +216,15 @@ initCardanoLedgerState pInfo = CardanoLedgerState
       { clsState = Consensus.pInfoInitLedger pInfo
       }
 
+-- TODO make this type safe. We make the assumption here that the first message of
+-- the chainsync protocol is 'RollbackTo'.
+readStateUnsafe :: LedgerEnv -> STM CardanoLedgerState
+readStateUnsafe env = do
+    mState <- readTVar $ leStateVar env
+    case mState of
+      Nothing -> panic "ledger state is not found"
+      Just st -> return st
```

Comment:
The rest of the code uses `pure` instead of `return` in preparation for https://gitlab.haskell.org/ghc/ghc/-/wikis/proposal/monad-of-no-return .